docs/src/design/stack/u32_ops.md

```diff
@@ -0,0 +1,294 @@
+
+To facilitate this operation, the prover sets values in $h_0, ..., h_3$ to 16-bit limbs of $a + b \cdot c$ with $h_0$ being the least significant limb. Thus, stack transition for this operation must satisfy the following constraints:
+
+>$$
+s_0 \cdot s_1 + s_2 = 2^{48} \cdot h_3 + 2^{32} \cdot h_2 + 2^{16} \cdot h_1 + h_0 \text{ | degree } = 2
+$$
+
+>$$
+s_1' = 2^{16} \cdot h_1 + h_0 \text{ | degree} = 1
+$$
+
+>$$
+s_0' = 2^{16} \cdot h_3 + h_2 \text{ | degree} = 1
+$$
+
+In addition to the above constraints, we also need to verify that values in $h_0, ..., h_3$ are smaller than $2^{16}$, which we can do using 16-bit range checks as described [previously](#range-checks). Also, we need to make sure that values in $h_0, ..., h_3$, when combined, form a valid field element, which we can do by putting a nondeterministic value $m$ into helper register $h_4$ and using the technique described [here](#checking-element-validity).
+
+**Note**: that the above constraints guarantee the correctness of the operation iff $a + b \cdot c$ cannot overflow field modules (which is the case for the field with modulus $2^{64} - 2^{32} + 1$).
+
+The effect of this operation on the rest of the stack is:
+* **Left shift** starting from position $3$.
+
+## U32DIV
+Assume $a$ and $b$ are the values at the top of the stack which are known to be smaller than $2^{32}$. The `U32DIV` operation computes $(c, d) \leftarrow a / b$, where $c$ contains the quotient and $d$ contains the remainder. The diagram below illustrates this graphically.
+
+![u32div](../../assets/design/stack/u32_operations/U32DIV.png)
+
+To facilitate this operation, the prover sets values in $h_0$ and $h_1$ to 16-bit limbs of $a - c$, and values in $h_2$ and $h_3$ to 16-bit limbs of $b - d - 1$. Thus, stack transition for this operation must satisfy the following constraints:
+
+>$$
+s_1 = s_0 \cdot s_1' + s_0' \text{ | degree } = 2
+$$
+
+>$$
+s_1 - s_1' = 2^{16} \cdot h_1 + h_0 \text{ | degree} = 1
+$$
+
+>$$
+s_0 - s_0' - 1= 2^{16} \cdot h_2 + h_3 \text{ | degree} = 1
+$$
+
+The second constraint enforces that $s_1' \leq s_1$, while the third constraint enforces that $s_0' < s_0$.
+
+The effect of this operation on the rest of the stack is:
+* **No change** starting from position $2$.
+
+## U32AND
+Assume $a$ and $b$ are the values at the top of the stack. The `U32AND` operation computes $c \leftarrow (a \land b)$, where $c$ is the result of performing a bitwise AND on $a$ and $b$. The diagram below illustrates this graphically.
+
+![u32and](../../assets/design/stack/u32_operations/U32AND.png)
+
+To facilitate this operation, we will need to make a request to the chiplet bus $b_{chip}$ by dividing its current value by the value representing bitwise operation request. This can be enforced with the following constraint:
+
+>$$
+b_{chip}' \cdot \left(\alpha_0 + \alpha_1 \cdot op_{u32and} + \alpha_2 \cdot s_0 + \alpha_3 \cdot s_1 +  \alpha_4 \cdot s_0'  \right) = b_{chip} \text{ | degree } = 2
+$$
```

Comment:
I believe we've previously described this as shown below, which would have the values of a and b swapped from what's described / pictured here. (I just want to make sure we're being consistent between here and the Bitwise chiplet). Same comment for `U32OR` and `U32XOR`.

$$
b_{chip}' \cdot \left(\alpha_0 + \alpha_1 \cdot op_{u32and} + \alpha_2 \cdot a + \alpha_3 \cdot b +  \alpha_4 \cdot c  \right) = b_{chip} \text{ | degree } = 2
$$

Reply:
Yep - I've updated the diagrams to be consistent with the above formula.